### PR TITLE
common: audit OPT_INT/OPT_UINT and handle incorrect signed uint

### PR DESCRIPTION
**Pull request by oliveiradan**
Changes proposed to address issue: BUG #14934 
http://tracker.ceph.com/issues/14934

### CODE REVIEW

**Comment by dachary** [comment]
needs rebasing

**Comment by liewegas** [comment]
needs rebase

File: src/ceph_mon.cc [hunk 1]
```diff
@@ -480,7 +480,7 @@ int main(int argc, const char **argv)
            << std::endl;
       exit(-err);
     }
-    if (stats.avail_percent <= g_conf->mon_data_avail_crit) {
+    if (static_cast<uint32_t>(stats.avail_percent) <= g_conf->mon_data_avail_crit) {
```
Comment (amitkumar50) [inline src/ceph_mon.cc]: @oliveiradan Please keep PR title < 50 characters
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

File: src/common/config.cc [hunk 2]
```diff
@@ -946,83 +952,177 @@ int md_config_t::set_val_impl(const char *val, const config_option *opt)
 
 int md_config_t::set_val_raw(const char *val, const config_option *opt)
 {
+  const int CEPH_SETTING_SAFE = 1;
```
Comment (amitkumar50) [inline src/common/config.cc]: @oliveiradan Why local variable is taken as caps?
Conventionally we take macros as this?

Comment (oliveiradan) [inline src/common/config.cc]: @amitkumar50 , that was when I started working on Ceph (that's an old PR) and I used it as a way to identify a constant. However, as it ended up getting old and a rebase was needed, I ended up not touching it (once it is a pretty big change) 'till we were certain this PR would still be useful or not. Should this PR still be interesting, I will rebase and rework whatever is needed.

Thanks,

**Comment by liewegas** [comment]
This is obsolete with the new options framework.  What we probably want instead is to translate this into min/max values in options.cc

**Pull request closed without merging**